### ISSUE TITLE
LlamaFactory performance gap vs Aifsdk

### ISSUE THREAD

**Issue by flyhero99** [adamlu123/mini-web-agent#47]
- Major fault: in evaluation, the web judge model was mistakenly assigned as vllm ckpt, instead of o4-mini.
- After the correction, the re-launched training job (using 3.5k web traj + 2.2k science data) lead to 43.7% performance on om2w (previous: 49.3%; still 5.6% performance gap)
- Flash Linear Attention and pre-tokenization both added
- Time cost: 10.5h (using 2x4-H100 nodes) for training 3 epochs (1600 steps)
- Training setting: 49k max cutoff len; ZeRO-2; enable_liger_kernel; flash_attn: fa2；bf16; 8GPU * 4 gradient accumulation step = 32 global bs; 
- Eval: om2w_spb_vllm_sum10.yaml
- Scoring: src/miniswewebagent/run/benchmarks/om2w.py
- Webjudge: o4-mini

**Comment by flyhero99** [comment]
09/03 Update:

(1) Reproduced the previous best-performing ckpt by phitrain (50-step limit: ~51%) on OSC

(2) Compared training setting between phitrain vs LlamaFactory, the major difference should be: currently, LlamaFactory freezes the vision weights and only train the textual weights, while phitrain trains all model weights.

(3) Started a new run with web-only data (3.5k trajs), same maxlen (40960), all weight training. The training should be done in 10~11 hours and then run an evaluation for ~5 hours (slower than MSR cluster because OSC limits the progress concurrency).
